first_assignment.py

```diff
@@ -1,11 +1,91 @@
-from PyQt5.QtWidgets import QApplication, QMainWindow, QWidget, QVBoxLayout, QLineEdit
-from PyQt5.QtGui import QPalette, QColor
+from PyQt5.QtWidgets import (
+    QApplication, QMainWindow, QLabel, QPushButton, 
+    QCheckBox, QSlider, QComboBox, QWidget, 
+    QHBoxLayout, QVBoxLayout, QLineEdit
+)
+from PyQt5.QtCore import Qt
 
 class MainWindow(QMainWindow):
+    """Custom window meant to display a textbox, dropdown bar,
+    yes/no checkboxes, slider from 0 to 100, and save button when
+    viewed from top to bottom.
+    """
     def __init__(self):
+        """Window constructor."""
+        # Declare instance variables required for YAML file
+        self.text = ""
```

Comment:
No need to declare these here

If you go this route you'd also have to connect a signal to a function that will set these whenever the the user changes the input slightly (bad), or have the submit button update these values once its pressed(better). 

Best way imo would be to build a dictionary of all the values grabbed from the ui whenever submit is pressed without storing these here